### PR TITLE
helm: Replace wait-for-it with a busybox script 

### LINKED ISSUE #24897: SPIRE initContainer with wait-for-it image needs to be updated for more CPU arches
### Description 

> Also comment seems lost, have we confirmed it runs on arm64 and amd64v1?

_Originally posted by @meyskens in https://github.com/cilium/cilium/pull/24765#discussion_r1164411976_

This issue to to verify the above comment to confirm if the image `cgr.dev/chainguard/wait-for-it@sha256:ecb58e3a2ffbdb732bb9049987e06eaf826d945410e167f31d6ffe28fab259f4` is working for both arm64 and arm64v1. 

In the worst case scenario, we should find one alternative image and update  respective value in helm chart.

### PR DESCRIPTION
**Pull request by meyskens**
Builds on  https://github.com/cilium/cilium/pull/24765

The wait-for-it image has problems in supporting some CPUs.
The image has a very simple functionality that is only needed to
avoid the spire agent to go into a crash loop.
This we recreated in busybox an image already present in our
setup for spire. This also avoids adding new dependencies.

Fixes: https://github.com/cilium/cilium/issues/24897

```release-note
Replace wait-for-it in SPIRE setup with a busybox script 
```

### CODE REVIEW

**Comment by sayboras** [comment]
last commit is really nice, I think you can rebase and mark for review :D

**Review by sayboras — APPROVED** [review]
While the changes in this PR look simple, there are a lot of thinking and knowledge embedded on the _why_. Thanks for always helping me 🦃 🎖️

**Comment by meyskens** [comment]
Test seems to fail on a non related issue
```
 	 The Service "echo-b" is invalid: spec.ports[0].nodePort: Invalid value: 31414: provided port is already allocated
```

File: install/kubernetes/cilium/templates/spire/agent/daemonset.yaml [hunk 1]
```diff
@@ -27,10 +27,14 @@ spec:
       hostNetwork: true
       dnsPolicy: ClusterFirstWithHostNet
       serviceAccountName: {{ .Values.auth.mTLS.spire.install.agent.serviceAccount.name }}
-      {{- if gt (len .Values.auth.mTLS.spire.install.agent.initContainers) 0 }}
       initContainers:
-        {{- toYaml .Values.auth.mTLS.spire.install.agent.initContainers | nindent 8 }}
-      {{- end }}
+        - name: init
+          image: docker.io/library/busybox:1.35.0@sha256:223ae047b1065bd069aac01ae3ac8088b3ca4a527827e283b85112f29385fb1b
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/spire/agent/daemonset.yaml:32]: should we have Renovate bot setup to update this image?

**Review by kaworu — APPROVED** [review]
Thanks for the PR @meyskens!

> Test seems to fail on a non related issue
> 
> ```
>  	 The Service "echo-b" is invalid: spec.ports[0].nodePort: Invalid value: 31414: provided port is already allocated
> ```

Please take the time to search for a `ci/flake` issue so we can track how many time we hit them. Here CI hit https://github.com/cilium/cilium/issues/13071.

**Comment by kaworu** [comment]
/test-1.26-4.19

File: install/kubernetes/cilium/templates/spire/agent/daemonset.yaml (hunk 1 above)
Comment (meyskens) [reply install/kubernetes/cilium/templates/spire/agent/daemonset.yaml:32]: Added a rule :)

File: .github/renovate.json5 [hunk 2]
```diff
@@ -181,6 +182,15 @@
         "v1.11"
       ]
     },
+    {
+      "matchPackageNames": [
+        "docker.io/library/busybox"
+      ],
+      "allowedVersions": "<1.35",
```
Comment (kaworu) [inline .github/renovate.json5]: Should this be at least `<=1.35`? My understanding is that as-is Renovate will never update the image.

**Review by joamaki — APPROVED** [review]

File: install/kubernetes/cilium/templates/spire/agent/daemonset.yaml [hunk 3]
```diff
@@ -27,10 +27,14 @@ spec:
       hostNetwork: true
       dnsPolicy: ClusterFirstWithHostNet
       serviceAccountName: {{ .Values.auth.mTLS.spire.install.agent.serviceAccount.name }}
-      {{- if gt (len .Values.auth.mTLS.spire.install.agent.initContainers) 0 }}
       initContainers:
-        {{- toYaml .Values.auth.mTLS.spire.install.agent.initContainers | nindent 8 }}
-      {{- end }}
+        - name: init
```
Comment (bimmlerd) [inline install/kubernetes/cilium/templates/spire/agent/daemonset.yaml:31]: so afaik changing `initContainers` is the thing that breaks users who don't follow proper upgrade methods, and instead just bump their image. I think the conclusion of talking about this recently was that if this needs to happen, it needs to be announced as loudly as possible.
cc @michi-covalent where do we announce this?

File: .github/renovate.json5 [hunk 4]
```diff
@@ -170,7 +171,7 @@
         "main",
         "v1.13"
       ]
-    },
+    },
```
Comment (bimmlerd) [inline .github/renovate.json5]: nit: spurious whitespace

**Review by bimmlerd — COMMENTED** [review]
Let's find a way to be loud about the `initContainer` change - I think at least we need to make it more explicit in the release note.

File: install/kubernetes/cilium/templates/spire/agent/daemonset.yaml (hunk 3 above)
Comment (meyskens) [reply install/kubernetes/cilium/templates/spire/agent/daemonset.yaml:31]: This got added only in main not even in a snapshot release to this should not be an issue in the upgrade flow

File: .github/renovate.json5 (hunk 4 above)
Comment (meyskens) [reply .github/renovate.json5]: fixed

File: .github/renovate.json5 (hunk 2 above)
Comment (meyskens) [reply .github/renovate.json5]: fixed

File: install/kubernetes/cilium/templates/spire/agent/daemonset.yaml (hunk 3 above)
Comment (bimmlerd) [reply install/kubernetes/cilium/templates/spire/agent/daemonset.yaml:31]: ah I didn't realise, fine by me then

**Review by bimmlerd — APPROVED** [review]

**Comment by meyskens** [comment]
Sorry for the last push, did a commit on the wrong branch

**Comment by meyskens** [comment]
/test-k8s-1.27-kernel-net-next

(build provision failed)

**Comment by meyskens** [comment]
/test-k8s-1.27-kernel-net-next

**Comment by meyskens** [comment]
Rebased for 1.27 revert on main

**Comment by meyskens** [comment]
runtime Ci failure hit https://github.com/cilium/cilium/issues/25178

**Pull request merged**